etc/dbus-serialbattery/battery.py

```diff
@@ -275,6 +275,9 @@ def manage_charge_voltage_linear(self) -> None:
                     and self.allow_max_voltage
                 ):
                     self.max_voltage_start_time = int(time())
+                    # Assume battery SOC ist 100% at this stage
+                    # Function for each battery that supports reset of SOC
+                    self.trigger_soc_reset()
```

Comment:
This is the wrong place. You have to put it there, when it switches from bulk/absorption to float. After line 292.